ts/IdentificationModal.tsx

```diff
@@ -128,18 +139,117 @@ class IdentificationModal extends React.PureComponent<Props, State> {
     };
   }
 
+  public componentDidMount() {
+    AppState.addEventListener("change", this.appStateChangeListener.bind(this));
+  }
+
+  public componentDidUnMount() {
+    AppState.removeEventListener(
+      "change",
+      this.appStateChangeListener.bind(this)
+    );
+  }
+
   public componentWillMount() {
-    TouchID.isSupported(isSupportedConfig)
-      .then(
-        biometryType => (biometryType === true ? "Fingerprint" : biometryType),
-        _ => undefined
-      )
-      .then(biometryType => this.setState({ biometryType }), _ => 0);
+    const { isFingerprintEnabled } = this.props;
+
+    if (isFingerprintEnabled) {
+      TouchID.isSupported(isSupportedConfig)
+        .then(
+          biometryType =>
+            biometryType === true ? "Fingerprint" : biometryType,
+          _ => undefined
+        )
+        .then(biometryType => this.setState({ biometryType }), _ => 0);
+    }
   }
 
-  public render() {
+  /**
+   * Check if fingerprint login can be prompted by looking at two parameters in
+   * serie:
+   * 1. `isFingerprintEnabled` coming from global state persisted preferences
+   * 2. The current status of biometry recognition support provided by querying
+   * the library in charge.
+   *
+   * It gets `updateBiometrySupportProp` as a boolean because it can be run
+   * from several contexts. When getting called while the app is returning
+   * foreground from background, biometry support has to be updated if system
+   * preferences changes happened.
+   *
+   * @param updateBiometrySupportProp
+   */
+  private maybeTriggerFingerprintRequest(
+    updateBiometrySupportProp?: updateBiometrySupportPropType
+  ) {
+    const { identificationState, isFingerprintEnabled } = this.props;
+
+    if (identificationState.kind !== "started") {
+      return;
+    }
+
+    // Check for global properties to know if biometric recognition is enabled
+    if (isFingerprintEnabled) {
+      TouchID.isSupported(isSupportedConfig)
```

Comment:
this looks like the same code al row 157, perhaps extract to a utility function?

Reply:
I've done some edits to increase code reuse. If it's not enought I was thinking of separating logics in two parts: `isFingerprintAvailable` and `isFingerprintAvailableAndUpdateState`. They both return a Promise that is used to trigger the fingerprint modal.